fuzzyfinder/main.py

```diff
@@ -17,10 +17,12 @@ def fuzzyfinder(input, collection, accessor=lambda x: x):
     suggestions = []
     input = str(input) if not isinstance(input, str) else input
     pat = '.*?'.join(map(re.escape, input))
+    pat = '(?=({0}))'.format(pat)   # lookahead regex to manage overlapping matches
     regex = re.compile(pat, re.IGNORECASE)
     for item in collection:
-        r = regex.search(accessor(item))
+        r = list(regex.finditer(accessor(item)))
         if r:
-            suggestions.append((len(r.group()), r.start(), accessor(item), item))
+            best = min(r, key=lambda x: len(x.group(1)))   # find shortest match
+            suggestions.append((len(best.group(1)), best.start(), accessor(item), item))
```

Comment:
It was already using the length of the match to rank the entries. I'm not sure what you gain by adding the lookahead matching. 

Try your test case without the changes you have made and you'll see that it produces the right results.

Reply:
Sorry, my test case itself was wrong. What I meant was this:

```python
import re

regex = re.compile('.*?'.join('fuz'))

regex.search('fufuz')
```

`fufuz` is matched instead of `fuz`, because `fufuz` overlaps `fuz`.

This is the error produced when running the test with the original `fuzzyfinder`:

```python
========================================================== FAILURES ==========================================================
_________________________________________ test_use_shortest_match_if_matches_overlap _________________________________________

    def test_use_shortest_match_if_matches_overlap():
        collection_list = ['fuuz', 'fuz', 'fufuz']
        text = 'fuz'
        results = fuzzyfinder(text, collection_list)
        expected = ['fuz', 'fufuz', 'fuuz']
>       assert list(results) == expected
E       AssertionError: assert ['fuz', 'fuuz', 'fufuz'] == ['fuz', 'fufuz', 'fuuz']
E         At index 1 diff: 'fuuz' != 'fufuz'
E         Use -v to get the full diff
```